perfkitbenchmarker/providers/ibmcloud/ibmcloud_virtual_machine.py

```diff
@@ -0,0 +1,599 @@
+    """Create a VM's scratch disk.
+
+    Args:
+      disk_spec: virtual_machine.BaseDiskSpec object of the disk.
+    """
+    disks_names = ('%s-data-%d-%d'
+                   % (self.name, len(self.scratch_disks), i)
+                   for i in range(disk_spec.num_striped_disks))
+    disks = [ibmcloud_disk.IbmCloudDisk(disk_spec, name, self.zone, \
+                                        encryption_key=self.data_encryption_key)
+             for name in disks_names]
+
+    self._CreateScratchDiskFromDisks(disk_spec, disks)
+
+  def DownloadPreprovisionedData(self, install_path, module_name, filename):
+    """Creats a temp file, no download """
+    self.RemoteCommand('echo "1234567890" > ' + \
+                       os.path.join(install_path, filename))
+
+  def ShouldDownloadPreprovisionedData(self, module_name, filename):
+    """Returns whether or not preprovisioned data is available."""
+    return True
+
+
+class DebianBasedIbmCloudVirtualMachine(IbmCloudVirtualMachine,
+                                        linux_virtual_machine.BaseDebianMixin):
+
+  def PrepareVMEnvironment(self):
+    logging.info('Pausing for 2 min before update and installs')
+    time.sleep(_WAIT_TIME_DEBIAN)
+    self.RemoteCommand('DEBIAN_FRONTEND=noninteractive apt-get -y update')
+    self.RemoteCommand('DEBIAN_FRONTEND=noninteractive apt-get -y install sudo')
+    super(DebianBasedIbmCloudVirtualMachine, self).PrepareVMEnvironment()
+
+
+class Debian9BasedIbmCloudVirtualMachine(DebianBasedIbmCloudVirtualMachine,
+                                         linux_virtual_machine.Debian9Mixin):
+  IMAGE_NAME_PREFIX = 'ibm-debian-9-'
+
+
+class Debian10BasedIbmCloudVirtualMachine(DebianBasedIbmCloudVirtualMachine,
+                                          linux_virtual_machine.Debian10Mixin):
+  IMAGE_NAME_PREFIX = 'ibm-debian-10-'
+
+
+class Ubuntu1604BasedIbmCloudVirtualMachine(IbmCloudVirtualMachine,
+                                            linux_virtual_machine.Ubuntu1604Mixin):
+  IMAGE_NAME_PREFIX = 'ibm-ubuntu-16-04-'
+
+  def PrepareVMEnvironment(self):
+    self.RemoteCommand('DEBIAN_FRONTEND=noninteractive apt-get -y update')
+    super(Ubuntu1604BasedIbmCloudVirtualMachine, self).PrepareVMEnvironment()
+
+
+class Ubuntu1804BasedIbmCloudVirtualMachine(IbmCloudVirtualMachine,
+                                            linux_virtual_machine.Ubuntu1804Mixin):
+  IMAGE_NAME_PREFIX = 'ibm-ubuntu-18-04-'
+
+  def PrepareVMEnvironment(self):
+    logging.info('Pausing for 10 min before update and installs')
+    time.sleep(_WAIT_TIME_UBUNTU)
+    self.RemoteCommand('DEBIAN_FRONTEND=noninteractive apt-get -y update')
+    super(Ubuntu1804BasedIbmCloudVirtualMachine, self).PrepareVMEnvironment()
+
+
+class RhelBasedIbmCloudVirtualMachine(IbmCloudVirtualMachine,
+                                      linux_virtual_machine.BaseRhelMixin):
+
+  def PrepareVMEnvironment(self):
+    logging.info('Pausing for 5 min before update and installs')
```

Comment:
This looks a bit odd.  Is there some other way of doing this other than waiting 5 minutes?

Reply:
redhat needs some time to register with subscription management server before tools can be installed properly, this process has however has improved, 1 min should be good.